lib/domain.js

```diff
@@ -85,15 +85,22 @@ const asyncHook = createHook({
 // another one. The stack is each entered domain.
 const stack = [];
 exports._stack = stack;
-process._setupDomainUse(stack);
+process._setupDomainUse();
 
-class Domain extends EventEmitter {
+function updateShouldAbort() {
+  process.shouldAbortOnUncaughtException =
+      stack.every((domain) => domain.listenerCount('error') === 0);
```

Comment:
Also, realize this isn't necessarily perf sensitive code but we could just abstract this to a separate top scope function like `const checkErrorListenerCount = (domain) => domain.listenerCount('error') === 0`. Take it or leave it, this is a very, very minor nit.

(The main impact of a change like this would be on future contributors more than practical impacts on this specific code.)

Reply:
I would expect the perf to be the same, this function doesn’t actually close over any outside variables.

Reply:
I've tested a similar situation before and it's about 10% difference. I would think it's because the arrow function has to be created separately each time (I don't imagine this is something that V8 can optimize) but it's possible there's another reason that I overlooked.

But as I said, it's not perf sensitive so this was more about steering future contributors in the right direction as far as these type of things go.